--- less-394/configure.ac~	2006-02-23 01:51:08.000000000 +0200
+++ less-394/configure.ac	2006-02-23 01:51:54.000000000 +0200
@@ -23,10 +23,10 @@
 AC_SYS_LARGEFILE
 
 # Checks for general libraries.
-AC_CHECK_LIB(xcurses, initscr, [have_xcurses=yes], [have_xcurses=no])
-AC_CHECK_LIB(ncursesw, initscr, [have_ncursesw=yes], [have_ncursesw=no])
-AC_CHECK_LIB(ncurses, initscr, [have_ncurses=yes], [have_ncurses=no])
-AC_CHECK_LIB(curses, initscr, [have_curses=yes], [have_curses=no])
+AC_CHECK_LIB(xcurses, tgetent, [have_xcurses=yes], [have_xcurses=no])
+AC_CHECK_LIB(ncursesw, tgetent, [have_ncursesw=yes], [have_ncursesw=no])
+AC_CHECK_LIB(ncurses, tgetent, [have_ncurses=yes], [have_ncurses=no])
+AC_CHECK_LIB(curses, tgetent, [have_curses=yes], [have_curses=no])
 AC_CHECK_LIB(termcap, tgetent, [have_termcap=yes], [have_termcap=no])
 AC_CHECK_LIB(termlib, tgetent, [have_termlib=yes], [have_termlib=no])
 # Regular expressions (regcmp) are in -lgen on Solaris 2,
