--- less-481/filename.c.orig	2015-10-30 20:54:28.100188419 +0100
+++ less-481/filename.c	2015-10-30 21:16:14.293494525 +0100
@@ -582,37 +582,7 @@
 {
 	FILE *fd;
 
-#if HAVE_SHELL
-	char *shell;
-
-	shell = lgetenv("SHELL");
-	if (shell != NULL && *shell != '\0')
-	{
-		char *scmd;
-		char *esccmd;
-
-		/*
-		 * Read the output of <$SHELL -c cmd>.  
-		 * Escape any metacharacters in the command.
-		 */
-		esccmd = shell_quote(cmd);
-		if (esccmd == NULL)
-		{
-			fd = popen(cmd, "r");
-		} else
-		{
-			int len = (int) (strlen(shell) + strlen(esccmd) + 5);
-			scmd = (char *) ecalloc(len, sizeof(char));
-			SNPRINTF3(scmd, len, "%s %s %s", shell, shell_coption(), esccmd);
-			free(esccmd);
-			fd = popen(scmd, "r");
-			free(scmd);
-		}
-	} else
-#endif
-	{
 		fd = popen(cmd, "r");
-	}
 	/*
 	 * Redirection in `popen' might have messed with the
 	 * standard devices.  Restore binary input mode.
