--- less-661/filename.c.orig	2024-08-24 20:24:56.143727073 +0200
+++ less-661/filename.c	2024-08-24 20:27:41.742829946 +0200
@@ -550,37 +550,7 @@ static FILE * shellcmd(constant char *cm
 {
 	FILE *fd;
 
-#if HAVE_SHELL
-	constant char *shell;
-
-	shell = lgetenv("SHELL");
-	if (!isnullenv(shell))
-	{
-		char *scmd;
-		char *esccmd;
-
-		/*
-		 * Read the output of <$SHELL -c cmd>.  
-		 * Escape any metacharacters in the command.
-		 */
-		esccmd = shell_quote(cmd);
-		if (esccmd == NULL)
-		{
-			fd = popen(cmd, "r");
-		} else
-		{
-			size_t len = strlen(shell) + strlen(esccmd) + 5;
-			scmd = (char *) ecalloc(len, sizeof(char));
-			SNPRINTF3(scmd, len, "%s %s %s", shell, shell_coption(), esccmd);
-			free(esccmd);
-			fd = popen(scmd, "r");
-			free(scmd);
-		}
-	} else
-#endif
-	{
 		fd = popen(cmd, "r");
-	}
 	/*
 	 * Redirection in `popen' might have messed with the
 	 * standard devices.  Restore binary input mode.
