diff -ruNp less-608.orig/ch.c less-608/ch.c
--- less-608.orig/ch.c	2022-07-22 21:26:24.000000000 +0200
+++ less-608/ch.c	2023-02-12 02:22:29.484707848 +0100
@@ -28,6 +28,7 @@ extern ino_t curr_ino;
 
 #if HAVE_PROCFS
 #include <sys/statfs.h>
+#include <linux/magic.h>
 #endif
 
 typedef POSITION BLOCKNUM;
diff -ruNp less-608.orig/configure.ac less-608/configure.ac
--- less-608.orig/configure.ac	2022-07-22 21:26:24.000000000 +0200
+++ less-608/configure.ac	2023-02-12 02:22:29.484707848 +0100
@@ -271,7 +271,8 @@ AC_TRY_COMPILE([#include <sys/types.h>
   [struct stat s; dev_t dev = s.st_dev; ino_t ino = s.st_ino;],
   [AC_MSG_RESULT(yes); AC_DEFINE(HAVE_STAT_INO)], [AC_MSG_RESULT(no)])
 AC_MSG_CHECKING(for procfs)
-AC_TRY_COMPILE([#include <sys/statfs.h>],
+AC_TRY_COMPILE([#include <sys/statfs.h>
+#include <linux/magic.h>],
   [struct statfs s; s.f_type = PROC_SUPER_MAGIC; (void) fstatfs(0,&s); ],
   [AC_MSG_RESULT(yes); AC_DEFINE(HAVE_PROCFS)], [AC_MSG_RESULT(no)])
 
