--- lesspipe.sh.orig	Sat Jun 23 07:05:30 2001
+++ lesspipe.sh	Tue Jun 26 02:54:46 2001
@@ -26,21 +26,21 @@
 
 lesspipe() {
   case "$1" in
-  *.tar) tar tvvf $1 2>/dev/null ;; # View contents of .tar and .tgz files
-  *.tgz) tar tzvvf $1 2>/dev/null ;;
-  *.tar.gz) tar tzvvf $1 2>/dev/null ;;
-  *.tar.Z) tar tzvvf $1 2>/dev/null ;;
-  *.tar.z) tar tzvvf $1 2>/dev/null ;;
-  *.tar.bz2) tar tyvvf $1 2>/dev/null ;;
-  *.Z) gzip -dc $1  2>/dev/null ;; # View compressed files correctly
-  *.z) gzip -dc $1  2>/dev/null ;;
-  *.gz) gzip -dc $1  2>/dev/null ;;
-  *.bz2) bzip2 -dc $1 2>/dev/null ;;
-  *.zip) unzip -l $1 2>/dev/null ;;
-  *.1|*.2|*.3|*.4|*.5|*.6|*.7|*.8|*.9|*.n|*.man) FILE=`file -L $1` ; # groff src
+  *.tar) tar tvvf "$1" 2>/dev/null ;; # View contents of .tar and .tgz files
+  *.tgz) tar tzvvf "$1" 2>/dev/null ;;
+  *.tar.gz) tar tzvvf "$1" 2>/dev/null ;;
+  *.tar.Z) tar tzvvf "$1" 2>/dev/null ;;
+  *.tar.z) tar tzvvf "$1" 2>/dev/null ;;
+  *.tar.bz2) tar tyvvf "$1" 2>/dev/null ;;
+  *.Z) gzip -dc "$1" 2>/dev/null ;; # View compressed files correctly
+  *.z) gzip -dc "$1" 2>/dev/null ;;
+  *.gz) gzip -dc "$1" 2>/dev/null ;;
+  *.bz2) bzip2 -dc "$1" 2>/dev/null ;;
+  *.zip) unzip -l "$1" 2>/dev/null ;;
+  *.1|*.2|*.3|*.4|*.5|*.6|*.7|*.8|*.9|*.n|*.man) FILE=`file -L "$1"` ; # groff src
     FILE=`echo $FILE | cut -d ' ' -f 2`
     if [ "$FILE" = "troff" ]; then
-      groff -s -p -t -e -Tascii -mandoc $1
+      groff -s -p -t -e -Tascii -mandoc "$1"
     fi ;;
 #  *) FILE=`file -L $1` ; # Check to see if binary, if so -- view with 'strings'
 #    FILE1=`echo $FILE | cut -d ' ' -f 2`
@@ -52,4 +52,4 @@
   esac
 }
 
-lesspipe $1
+lesspipe "$1"
