--- lesso/configure.ac	Sat Jan 18 05:55:15 2003
+++ less/configure.ac	Sun Feb 16 22:28:19 2003
@@ -23,9 +23,9 @@
 AC_SYS_LARGEFILE
 
 # Checks for general libraries.
-AC_CHECK_LIB(xcurses, initscr, [have_xcurses=yes], [have_xcurses=no])
-AC_CHECK_LIB(ncurses, initscr, [have_ncurses=yes], [have_ncurses=no])
-AC_CHECK_LIB(curses, initscr, [have_curses=yes], [have_curses=no])
+AC_CHECK_LIB(xcurses, tgetent, [have_xcurses=yes], [have_xcurses=no])
+AC_CHECK_LIB(ncurses, tgetent, [have_ncurses=yes], [have_ncurses=no])
+AC_CHECK_LIB(curses, tgetent, [have_curses=yes], [have_curses=no])
 AC_CHECK_LIB(termcap, tgetent, [have_termcap=yes], [have_termcap=no])
 AC_CHECK_LIB(termlib, tgetent, [have_termlib=yes], [have_termlib=no])
 # Regular expressions (regcmp) are in -lgen on Solaris 2,
